### ISSUE TITLE
Investigate why test_corner_case.cc does not fail

### ISSUE THREAD

**Issue by nicolasvasilache** [facebookresearch/TensorComprehensions#397]
This test looks wrong to me: the inferred output is 1-D but it passes with a 2-D output
Investigate and address:

```
TEST(TestCornerCases, E10) {
  Succeed(
      "def f(int32 a) -> (b) { b(i) = a where i in 0:10 }", {I()}, {I(10, 10)});
}
```

**Comment by nicolasvasilache** [comment]
it's related to the impl in the old API; the new API catches this properly as an error so I fixed the test in #379

**Comment by nicolasvasilache** [comment]
Actually let's add a negative case that fails once the new API has landed.
